Dockerfile

```diff
@@ -1,40 +1,73 @@
-FROM node:10-buster-slim
-LABEL maintainer="HMPPS Digital Studio <info@digital.justice.gov.uk>"
+# Build stage 1.
+# Create base image with python (required for node-gyp)
 ARG BUILD_NUMBER
 ARG GIT_REF
 
-RUN apt-get update && \
-    apt-get upgrade -y && \
-    apt-get install -y make python curl wget && \
-    apt-get autoremove -y && \
-    rm -rf /var/lib/apt/lists/*
+FROM node:10-buster-slim as base
+
+LABEL maintainer="HMPPS Digital Studio <info@digital.justice.gov.uk>"
 
 ENV TZ=Europe/London
 RUN ln -snf "/usr/share/zoneinfo/$TZ" /etc/localtime && echo "$TZ" > /etc/timezone
 
 RUN addgroup --gid 2000 --system appgroup && \
     adduser --uid 2000 --system appuser --gid 2000
 
-# Create app directory
-RUN mkdir -p /app
 WORKDIR /app
 
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y make python
+
+# Build stage 2.
+# This stage builds our assets.
+FROM base as build
+ARG BUILD_NUMBER
+ARG GIT_REF
+
+RUN apt-get install -y curl wget
+
 # Install AWS RDS Root cert
 RUN mkdir /home/appuser/.postgresql \
   && curl https://s3.amazonaws.com/rds-downloads/rds-ca-2019-root.pem \
-    > /app/root.cert        
+    > /app/root.cert
 
-COPY --chown=appuser:appgroup . .
+COPY . .
 
 ENV BUILD_NUMBER ${BUILD_NUMBER:-1_0_0}
 ENV GIT_REF ${GIT_REF:-dummy}
 
-RUN npm ci && \
-    npm run build && \
+# Remove cypress as it takes ~1 minute to download and install
+RUN sed -ie 's/"cypress.*//g' package.json
+
+RUN npm install && npm run build  && \
     export BUILD_NUMBER=${BUILD_NUMBER} && \
     export GIT_REF=${GIT_REF} && \
     npm run record-build-info
 
+# Build stage 3.
+# This stage builds the final Docker image that we'll use in production.
+FROM base
+
+RUN apt-get autoremove -y && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY --from=build --chown=appuser:appgroup \
+        /app/package.json \
+        /app/package-lock.json \
+        /app/dist \
+        /app/root.cert \
+        /app/build-info.json \
+        ./
+
+COPY --from=build --chown=appuser:appgroup \
+        /app/assets ./assets
+
+COPY --from=build --chown=appuser:appgroup \
+        /app/server/views ./server/views
+
+RUN npm ci --only=production
```

Comment:
This ends up doing npm install twice..

an alternative would be copying over `node_modules` and doing `npm prune --production` but this seems to take longer and also results in slightly larger docker images (though has the benefit of not downloading the internet again)